WEEK 1 NONGRADED CHALLENGE

Kamu adalah salah satu tim relawan back office penanganan COVID-19 di Indonesia. Kamu diminta untuk memperoleh data baru dari dataset yang sudah disediakan. Kerjakan dan jawab pertanyaan-pertanyaan berikut. (Dataset yang disediakan merupakan jumlah kasus kumulatif tiap harinya)

Data dapat diakses pada link berikut: https://raw.githubusercontent.com/fahmimnalfrzki/Dataset/main/data_covid_dunia_13_apr_22.csv

Task 1

Buatlah sebuah fungsi untuk menghitung penambahan jumlah kasus harian Covid-19 dari data kumulatif yang disediakan! (Hint: Kamu bisa menggunakan .diff() yang disediakan oleh Pandas untuk menghitung perbedaan tiap baris data/menghitung penambahan jumlah kasus harian dan perhatikan missing value setelah perhitungan kasus harian, isi missing value dengan angka 0!)

In [29]:
import numpy as np
import pandas as pd

In [30]:
df = pd.read_csv('data_covid_dunia_13_apr_22.csv')
covid = df.copy()

In [115]:
covid

Afghanistan  Albania  Algeria  Andorra  Angola  Antarctica  \
Year                                                                  
1/22/20            0        0        0        0       0           0   
1/23/20            0        0        0        0       0           0   
1/24/20            0        0        0        0       0           0   
1/25/20            0        0        0        0       0           0   
1/26/20            0        0        0        0       0           0   
...              ...      ...      ...      ...     ...         ...   
4/9/22        177974   274191   265720    40328   99194          11   
4/10/22       177974   274219   265724    40328   99194          11   
4/11/22       178141   274219   265727    40328   99194          11   
4/12/22       178257   274272   265730    40328   99194          11   
4/13/22       178295   274320   265731    40709   99194          11   

         Antigua and Barbuda  Argentina  Armenia  Australia  ...  Uruguay  \
Year                                                         ...            
1/22/20                    0          0        0          0  ...        0   
1/23/20                    0          0        0          0  ...        0   
1/24/20                    0          0        0          0  ...        0   
1/25/20                    0          0        0          0  ...        0   
1/26/20                    0          0        0          0  ...        0   
...                      ...        ...      ...        ...  ...      ...   
4/9/22                  7511    9052083   422662      86887  ...   893575   
4/10/22                 7511    9052536   422677      87634  ...   893909   
4/11/22                 7511    9054126   422678      88543  ...   894267   
4/12/22                 7523    9056203   422691      89510  ...   894630   
4/13/22                 7523    9057923   422711      90567  ...   894984   

         Uzbekistan  Vanuatu  Venezuela   Vietnam  West Bank and Gaza  \
Year                                                                    
1/22/20           0        0          0         0                   0   
1/23/20           0        0          0         2                   0   
1/24/20           0        0          0         2                   0   
1/25/20           0        0          0         2                   0   
1/26/20           0        0          0         2                   0   
...             ...      ...        ...       ...                 ...   
4/9/22       238040     5125     521373  10169929              656287   
4/10/22      238071     5480     521471  10198236              656287   
4/11/22      238088     5487     521471  10250160              656287   
4/12/22      238098     5536     521546  10272964              656617   
4/13/22      238145     5543     521714  10297587              656617   

         Winter Olympics 2022  Yemen  Zambia  Zimbabwe  
Year                                                    
1/22/20                     0      0       0         0  
1/23/20                     0      0       0         0  
1/24/20                     0      0       0         0  
1/25/20                     0      0       0         0  
1/26/20                     0      0       0         0  
...                       ...    ...     ...       ...  
4/9/22                    535  11813  317483    246925  
4/10/22                   535  11813  317483    246958  
4/11/22                   535  11814  317804    247010  
4/12/22                   535  11814  317804    247094  
4/13/22                   535  11815  318113    247160  

[813 rows x 284 columns]

In [32]:
covid.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 813 entries, 0 to 812
Columns: 285 entries, Unnamed: 0 to Zimbabwe
dtypes: int64(284), object(1)
memory usage: 1.8+ MB


In [33]:
covid.rename(columns={'Unnamed: 0': 'Year'}, inplace=True)
covid.head()

Year  Afghanistan  Albania  Algeria  Andorra  Angola  Antarctica  \
0  1/22/20            0        0        0        0       0           0   
1  1/23/20            0        0        0        0       0           0   
2  1/24/20            0        0        0        0       0           0   
3  1/25/20            0        0        0        0       0           0   
4  1/26/20            0        0        0        0       0           0   

   Antigua and Barbuda  Argentina  Armenia  ...  Uruguay  Uzbekistan  Vanuatu  \
0                    0          0        0  ...        0           0        0   
1                    0          0        0  ...        0           0        0   
2                    0          0        0  ...        0           0        0   
3                    0          0        0  ...        0           0        0   
4                    0          0        0  ...        0           0        0   

   Venezuela  Vietnam  West Bank and Gaza  Winter Olympics 2022  Yemen  \
0          0        0                   0                     0      0   
1          0        2                   0                     0      0   
2          0        2                   0                     0      0   
3          0        2                   0                     0      0   
4          0        2                   0                     0      0   

   Zambia  Zimbabwe  
0       0         0  
1       0         0  
2       0         0  
3       0         0  
4       0         0  

[5 rows x 285 columns]

In [120]:
#covid.set_index('Year', inplace=True)

In [122]:
def covid2 (covid):
    covid = covid.diff()
    covid.fillna(0, inplace=True)
    return (covid)

covid2(covid)



Afghanistan  Albania  Algeria  Andorra  Angola  Antarctica  \
Year                                                                  
1/22/20          0.0      0.0      0.0      0.0     0.0         0.0   
1/23/20          0.0      0.0      0.0      0.0     0.0         0.0   
1/24/20          0.0      0.0      0.0      0.0     0.0         0.0   
1/25/20          0.0      0.0      0.0      0.0     0.0         0.0   
1/26/20          0.0      0.0      0.0      0.0     0.0         0.0   
...              ...      ...      ...      ...     ...         ...   
4/9/22           0.0     55.0      6.0      0.0     0.0         0.0   
4/10/22          0.0     28.0      4.0      0.0     0.0         0.0   
4/11/22        167.0      0.0      3.0      0.0     0.0         0.0   
4/12/22        116.0     53.0      3.0      0.0     0.0         0.0   
4/13/22         38.0     48.0      1.0    381.0     0.0         0.0   

         Antigua and Barbuda  Argentina  Armenia  Australia  ...  Uruguay  \
Year                                                         ...            
1/22/20                  0.0        0.0      0.0        0.0  ...      0.0   
1/23/20                  0.0        0.0      0.0        0.0  ...      0.0   
1/24/20                  0.0        0.0      0.0        0.0  ...      0.0   
1/25/20                  0.0        0.0      0.0        0.0  ...      0.0   
1/26/20                  0.0        0.0      0.0        0.0  ...      0.0   
...                      ...        ...      ...        ...  ...      ...   
4/9/22                   0.0      840.0     19.0      785.0  ...    566.0   
4/10/22                  0.0      453.0     15.0      747.0  ...    334.0   
4/11/22                  0.0     1590.0      1.0      909.0  ...    358.0   
4/12/22                 12.0     2077.0     13.0      967.0  ...    363.0   
4/13/22                  0.0     1720.0     20.0     1057.0  ...    354.0   

         Uzbekistan  Vanuatu  Venezuela  Vietnam  West Bank and Gaza  \
Year                                                                   
1/22/20         0.0      0.0        0.0      0.0                 0.0   
1/23/20         0.0      0.0        0.0      2.0                 0.0   
1/24/20         0.0      0.0        0.0      0.0                 0.0   
1/25/20         0.0      0.0        0.0      0.0                 0.0   
1/26/20         0.0      0.0        0.0      0.0                 0.0   
...             ...      ...        ...      ...                 ...   
4/9/22         46.0    203.0       48.0  34140.0                 0.0   
4/10/22        31.0    355.0       98.0  28307.0                 0.0   
4/11/22        17.0      7.0        0.0  51924.0                 0.0   
4/12/22        10.0     49.0       75.0  22804.0               330.0   
4/13/22        47.0      7.0      168.0  24623.0                 0.0   

         Winter Olympics 2022  Yemen  Zambia  Zimbabwe  
Year                                                    
1/22/20                   0.0    0.0     0.0       0.0  
1/23/20                   0.0    0.0     0.0       0.0  
1/24/20                   0.0    0.0     0.0       0.0  
1/25/20                   0.0    0.0     0.0       0.0  
1/26/20                   0.0    0.0     0.0       0.0  
...                       ...    ...     ...       ...  
4/9/22                    0.0    1.0     0.0       0.0  
4/10/22                   0.0    0.0     0.0      33.0  
4/11/22                   0.0    1.0   321.0      52.0  
4/12/22                   0.0    0.0     0.0      84.0  
4/13/22                   0.0    1.0   309.0      66.0  

[813 rows x 284 columns]

Task 2

Buat variabel baru yang berisikan data penambahan jumlah kasus positif harian Covid-19 di dunia menggunakan fungsi yang telah kamu buat sebelumnya!

In [132]:
covid_harian = covid2(covid).copy()   
covid_harian['Indonesia']

Year
1/22/20       0.0
1/23/20       0.0
1/24/20       0.0
1/25/20       0.0
1/26/20       0.0
            ...  
4/9/22     1468.0
4/10/22    1071.0
4/11/22    1196.0
4/12/22    1455.0
4/13/22    1551.0
Name: Indonesia, Length: 813, dtype: float64

Task 3

Pada tanggal berapa puncak tertinggi kasus harian Covid-19 di Indonesia? bandingkan dengan tanggal 10 April 2022, apakah mengalami penurunan atau kenaikan jumlah kasus dibandingkan tanggal tersebut? (Kamu bisa gunakan pandas query untuk hal ini)

In [124]:
covid_harian['Indonesia'].sort_values(ascending=False)

Year
2/16/22    64718.0
2/17/22    63956.0
2/23/22    61488.0
2/18/22    59635.0
2/19/22    59384.0
            ...   
2/16/20        0.0
2/15/20        0.0
2/14/20        0.0
2/13/20        0.0
1/22/20        0.0
Name: Indonesia, Length: 813, dtype: float64

In [133]:
covid_harian.loc['4/10/22','Indonesia']

1071.0

Punca tertinggi penambahan kasus baru Covid-19 di Indonesia terjadi pada tanggal 16 Februari 2022 dengan penambahan sebesar 64.718 pasien. Jika dibandingkan dengan penambahan kasus Covid-19 pada tanggal 10 April 2022 yang sebesar 1071 pasien, penambahan kasus Covid-19 di tanggal 16 Februari 2022 mengalami penurunan yang sangat signifikan.

Task 4

Gunakan for loop dan conditional if untuk membuat sebuah list yang berisikan penambahan jumlah kasus harian Covid-19 dengan ketentuan hanya mengambil data Indonesia saja dan jumlah kasus di bawah 10000. Berapa jumlah hari dengan kasus di bawah 10000? (Gunakan for loop dan conditional if dalam hal ini)

In [140]:
covid_u_10000=[]

indonesia_covid = covid_harian['Indonesia']
for value in indonesia_covid:
    if value < 10000:
        covid_u_10000.append(value)

len(covid_u_10000)

661

Jumlah harian kasus Covid 19 namun dengan kasus baru di bawah 10000 sebesar 661 hari